packages/devtools_app/lib/src/shared/analytics/analytics_controller.dart

```diff
@@ -6,27 +6,54 @@ import 'dart:async';
 
 import 'package:flutter/foundation.dart';
 
-import '_analytics_controller_stub.dart'
-    if (dart.library.js_interop) '_analytics_controller_web.dart';
+import '../dtd_manager_extensions.dart';
+import '../globals.dart';
+
+import 'analytics.dart' as ga;
+
+Future<AnalyticsController> get analyticsController async {
+  if (_analyticsController != null) return _analyticsController!;
+
+  // TODO(https://github.com/flutter/devtools/issues/7083): when the legacy
+  // analytics are fully removed, this try-catch is unnecessary because we will
+  // get these values directly from [dtdManater], like how the consentMessage
+  // parameter is specified below.
+  var enabled = false;
+  var shouldShowConsentMessage = false;
+  try {
+    enabled = await ga.isAnalyticsEnabled();
+    shouldShowConsentMessage = await ga.shouldShowAnalyticsConsentMessage();
```

Comment:
Do we have any legal concerns about showing the consent message for UA even if it was the legacy analytics that reported we should be showing the consent message?

This would only happen if UA reports false under `ga.shouldShowAnalyticsConsentMessage` but legacy reports true since this function checks both analytics systems.

Reply:
This does not change any behavior from what we had before this PR. Should we instead stop checking the legacy value for this one? We don't show the legacy message anymore, so maybe we should just defer to the UA logic for when to show the message. WDYT?